lib/rspec-puppet/adapters.rb

```diff
@@ -0,0 +1,107 @@
+module RSpec::Puppet
+  module Adapters
+
+    class Base
+      def setup_puppet(example_group)
+        settings_map.each do |puppet_setting, rspec_setting|
+          set_setting(example_group, puppet_setting, rspec_setting)
+        end
+      end
+
+      def set_setting(example_group, puppet_setting, rspec_setting)
+        if example_group.respond_to?(rspec_setting)
+          value = example_group.send(rspec_setting)
+        else
+          value = RSpec.configuration.send(rspec_setting)
+        end
+        begin
+          Puppet[puppet_setting] = value
+        rescue ArgumentError
+          # TODO: this silently swallows errors when applying settings that the current
```

Comment:
It's not worth holding this up. If and when this is revisited, we might want to add a depreciation warning using https://github.com/rspec/rspec-core/blob/master/lib/rspec/core/warnings.rb ; those are only printed once, and supply infrastructure to trace where they originated.